src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -3529,3 +3529,14 @@ def _verify_hostname_binding(cmd, resource_group_name, name, hostname, slot=None
             verified_hostname_found = True
 
     return verified_hostname_found
+
+
+def _get_plan_by_name(client, plan):
+    plan_info = None
+    plan_list = client.app_service_plans.list()
+    if plan_list:
+        candidate_plans = list(filter(lambda x: x.name == plan, plan_list))
+        if candidate_plans:
+            plan_info = candidate_plans[0]
```

Comment:
Why choose the first one?

Reply:
If we are choosing based on the name, it shouldn't really matter if this is the first, second, etc! If the name matches, should be enough.

Reply:
Is it possible that multiple plans have same names but different configurations?

Reply:
Yes! And that can happen because we are not using the resource group that the web app is located anymore!

Reply:
It is possible that multiple plans can have same names under different resource groups. So when we consider only the first plan, then there is a possibility of assigning a plan from different RG than the user anticipated RG. So as per me the current implementation will not solve the problem 100%.
**proposed implementation:**  In the portal plans are populated based on the selected `AppService OS` type. I think we should consider the same implementation for CLI command as well.

Reply:
I see! So, whats the way you suggest implementing that, since we have no mandatory parameter that tells the OS type on the az webapp create command?

Reply:
yes we don't have a parameter to tell the type of OS. we should either add a parameter as `--plan-resource-group`  to know the plan RG name or `--is_linux` to know the OS type.
@panchagnula - please let us know your suggestion on it.

Reply:
I am happy with both of the implementations! I feel that in order to avoid any type of confusing, we should add the ```--plan-resource-group``` as a optional parameter, and if its not passed, we try to get from the same rg as web app is! What do you think?

Reply:
Added the implementation for --plan-resource-group, let me know what you think!

Reply:
@Guilhermeslucas first of all thanks for taking the initiative in starting this PR. Unfortunately at this point we cannot take your change. One big feedback we get regarding CLI commands is to keep them simple & not add new parameters since for a non Azure domain knowledge customer - the commands are overwhelming, hence in order to fix the original issue here, our team needed to come up with a better solution to fix this ,avoiding adding any additional parameters. I am closing this PR for now since we cannot take this  with the current approach. I see you are an MS employee please feel free to contact me via email if you need more details. Thank you.

Reply:
huuum, thats sad :( No problem, I understand! Thanks for the answer!